tasks/change-granule-collection-s3/src/index.ts

```diff
@@ -0,0 +1,432 @@
+      targetCollection.version
+    ),
+  };
+}
+
+/**
+ * Update the cmr objects to contain data adherant to the target granules they reflect
+ */
+async function updateCMRData(
+  targetGranules: Array<ValidGranuleRecord>,
+  cmrObjectsByGranuleId: { [key: string]: Object },
+  cmrFilesByGranuleId: Dictionary<ValidApiFile>,
+  config: EventConfig
+): Promise<{ [key: string]: Object }> {
+  const distEndpoint = getRequiredEnvVar('DISTRIBUTION_ENDPOINT');
+  const bucketTypes = Object.fromEntries(Object.values(config.buckets)
+    .map(({ name, type }) => [name, type]));
+  const cmrGranuleUrlType = get(config, 'cmrGranuleUrlType', 'both');
+  const distributionBucketMap = await fetchDistributionBucketMap();
+  const outputObjects: { [key: string]: Object } = {};
+  targetGranules.forEach((targetGranule) => {
+    const cmrFile = cmrFilesByGranuleId[targetGranule.granuleId];
+    const cmrObject = cmrObjectsByGranuleId[targetGranule.granuleId];
+    if (!(cmrFile && cmrObject)) {
+      outputObjects[targetGranule.granuleId] = {};
+    }
+    outputObjects[targetGranule.granuleId] = updateCmrFileCollections({
+      collection: config.targetCollection,
+      cmrFileName: cmrFile.key,
+      cmrObject,
+      files: (targetGranule as ValidGranuleRecord).files,
+      distEndpoint,
+      bucketTypes,
+      cmrGranuleUrlType,
+      distributionBucketMap,
+    });
+  });
+  return outputObjects;
+}
+
+/**
+ * Build a set of granules according to new collection
+ * New granules reference new collectionId as their collectionId
+ * files for new granules are updated according to new collection url_path
+ * file names are *not* updated
+ */
+function buildTargetGranules(
+  granules: Array<ValidGranuleRecord>,
+  config: EventConfig,
+  cmrObjects: { [key: string]: Object },
+  targetCollection: CollectionRecord
+): Array<ValidGranuleRecord> {
+  const bucketsConfig = new BucketsConfig(config.buckets);
+  const targetGranules: Array<ValidGranuleRecord> = [];
+  const granulesAndMetadata = granules.map(
+    async (granule) => updateGranuleMetadata(
+      granule,
+      bucketsConfig,
+      cmrObjects,
+      targetCollection
+    )
+  );
+  granulesAndMetadata.forEach((targetGranule) => {
+    targetGranules.push(targetGranule);
+  });
+  return targetGranules;
+}
+
+async function changeGranuleCollectionS3(event: ChangeCollectionsS3Event): Promise<Object> {
+  const config = event.config;
+  const s3MultipartChunksizeMb = config.s3MultipartChunksizeMb
+    ? config.s3MultipartChunksizeMb : Number(process.env.default_s3_multipart_chunksize_mb);
+
+  const chunkSize = s3MultipartChunksizeMb ? s3MultipartChunksizeMb * MB : undefined;
+  const targetCollection = await getCollection({
+    prefix: getRequiredEnvVar('stackName'),
+    collectionName: config.targetCollection.name,
+    collectionVersion: config.targetCollection.version,
+  });
+
+  log.debug(`change-granule-collection-s3 config: ${JSON.stringify(event.config)}`);
```

Comment:
Generic: we should consider doing a quick audit of the code once this gets near merge and make sure we have reasonable logging coverage to make support easier.